lib/bundler/definition.rb

```diff
@@ -533,7 +533,14 @@ def converge_sources
 
     def converge_dependencies
       (@dependencies + @locked_deps).each do |dep|
-        dep.source = sources.get(dep.source) if dep.source
+        locked_source = @locked_deps.select {|d| d.name == dep.name }.last
```

Comment:
I don't believe so. `name` is really the only accessible, applicable attribute of the `dependency` object.